telemetry/record.go

```diff
@@ -208,6 +209,33 @@ func (record *Record) toJSON() ([]byte, error) {
 	return jsonOptions.Marshal(payload)
 }
 
+// ToMap converts the data of a record to a map or slice of maps
+func (record *Record) ToMap(excludeTypes bool) (interface{}, error) {
```

Comment:
`ToMap` -> `ToLoggingMap` to make things explicit?

Reply:
also this method kind of don't belong in record.go file. Possible to move it to `logger.go` file?

Reply:
This could be used in other producers going forward. It transforms a record so I feel it belongs here.

Reply:
Updated to be specific to logger. Other producers should prefer proto.